aep/general/0203/aep.md.j2

```diff
@@ -1,5 +1,278 @@
+field in a resource message, or a field of a message farther down the tree.
+Notably, fields in response messages (a message which only ever acts as a
+return value to an RPC, usually ending in `Response`) **should not** be
+described as output only because this is already implied.
+
+Output only fields **may** be set to empty values if appropriate to the API.
+
+Potential use cases for output only fields (this is not an exhaustive list)
+are:
+
+- Create or update timestamps.
+- Derived or structured information based on original user input.
+- Properties of a resource assigned by the service which can not be altered.
+
+### Required
+
+The use of `REQUIRED` indicates that the field **must** be present (and set to
+a non-empty value) on the request or resource.
+
+A field **should** only be described as required if _either_:
+
+- It is a field on a resource that a user provides somewhere as input. In this
+  case, the resource is only valid if a "truthy" value is _stored_.
+  - When [creating](./standard-methods) the resource, a value **must** be
+    provided for the field on the create request.
+  - When [updating](./standard-methods) the resource, the user **may** omit the
+    field provided that the field is also absent from the field mask,
+    indicating no change to the field (otherwise it **must** be provided).
+- It is a field on a request message (a message that is an argument to an RPC,
+  with a name usually ending in `Request`). In this case, a value **must** be
+  provided as part of the request, and failure to do so **must** cause an error
+  (usually `INVALID_ARGUMENT`).
+
+We define the term "truthy" above as follows:
+
+- For primitives, values other than `0`, `0.0`, empty string/bytes, and `false`
+- For repeated fields maps, values with at least one entry
+- For messages, any message with at least one "truthy" field.
+
+Fields **should not** be described as required in order to signify:
+
+- A field which will always be present in a response.
+- A field which is conditionally required in some situations.
+- A field on any message (including messages that are resources) which is never
+  used as user input.
+
+**Note:** In most cases, empty values (such as `false` for booleans, `0` for
+integers, or the unspecified value for enums) are indistinguishable from unset
+values, and therefore setting a required field to a falsy value yields an
+error. A corollary to this is that a required boolean must be set to `true`.
+
+### Unordered List
+
+The use of `UNORDERED_LIST` on a repeated field of a resource indicates that
+the service does not guarantee the order of the items in the list.
+
+A field **should** be described as an unordered list if the service does not
+guarantee that the order of the elements in the list will match the order that
+the user sent, including a situation where the service will sort the list on
+the user's behalf.
+
+A resource with an unordered list **may** return the list in a stable order, or
```

Comment:
is this sentence necessary? it seems pretty clear what this annotation guarantees based on the last two paragraphs.

Reply:
Yeah, I can remove this.